### ISSUE TITLE
Running Pytest Produces ModuleNotFound and KeyError errors

### ISSUE THREAD

**Issue by GiaJordan** [Sage-Bionetworks/schematic#593]
**Describe the bug**
When running pytest in the environment with schematic installed, it yields `ModuleNotFoundError: No module named 'dotenv'`, as `python-dotenv` does not appear to be included in the installation package for the environment. 
When pytest is able to run, there are errors for 19 tests, 17 of which are `KeyError: 'synapse_config'`.  This is resolved after adding the path to the `.synapseConfig` file.

**To Reproduce**
1. Create a virtual environment and install `schematicpy`
2. Run `pytest -vs ./tests` in the virtual environment

**Expected behavior**
Expect necessary modules to be installed along with schematic, and the test config to have paths to `.synapseConfig` included

**Priority** (select one)
- [ ] Minor ⬇️

**Desktop (if applicable, please complete the following information):**
Windows 10

**Comment by GiaJordan** [comment]
Here is the issue for what we discussed a while back @milen-sage

**Comment by linglp** [comment]
This is similar to the issue that I am having with [PR592](https://github.com/Sage-Bionetworks/schematic/pull/592). When I first ran the tests, I also had error message:  `KeyError: 'synapse_config'`. After adding synapse_config as a key to my test_config.yml (and adding my .synapseConfig path), I could run all the tests locally without any errors. But when I pushed my changes to GitHub, the tests were still failing. 

Note: I set up my virtual environment by using `poetry shell` and `poetry install`.  The second time that I ran the tests locally, I could comment out `synapse_config` without running into any issues. (But the error on the github VMs persisted)
